nowriter/tmwire/tmwire_test.go

```diff
@@ -16,6 +17,22 @@ import (
 var legacy = tmlegacy.TMEncoderLegacy{}
 var pure = tmencoding.TMEncoderPure{}
 var dec = tmdecoding.TMDecoderPure{}
+var arrayRaw = tmarrayencoder.TMArrayEncoderUnlengthPure{}
+var arrayPrefixed = tmarrayencoder.TMArrayEncoderLengthPure{}
+
+func TestArrayRaw(t *testing.T) {
+	simpleTest := []uint16{0x1122, 0x3344}
+	expected := []byte{0x11, 0x22, 0x33, 0x44}
+	encodedBytes := arrayRaw.EncodeUint16Array(simpleTest)
+	assert.Zero(t, bytes.Compare(encodedBytes, expected))
+}
+
+func TestArrayPrefix(t *testing.T) {
+	simpleTest := []uint16{0x1122, 0x3344}
+	expected := []byte{0x1, 0x2, 0x11, 0x22, 0x33, 0x44}
+	encodedBytes := arrayPrefixed.EncodeUint16Array(simpleTest)
+	assert.Zero(t, bytes.Compare(encodedBytes, expected))
+}
```

Comment:
```
import "testing/quick"
...
func TestArray(t *testing.T) {                                                                                                           
  propertyTest := func(input []uint8) bool {                                                                                             
    truncatedInputLen := len(input) / 2 * 2                                                                                              
    truncatedInput := input[:truncatedInputLen]                                                                                          
                                                                                                                                         
    pairs := []uint16{}                                                                                                                  
    for i := 0; i+1 < truncatedInputLen; i += 2 {                                                                                        
      pair := uint16(truncatedInput[i]<<8) + uint16(truncatedInput[i+1])                                                                 
      pairs = append(pairs, pair)                                                                                                        
    }                                                                                                                                    
                                                                                                                                         
    encodedBytes := arrayRaw.EncodeUint16Array(pairs)                                                                                    
                                                                                                                                         
    return bytes.Compare(encodedBytes, input) == 0                                                                                       
  }                                                                                                                                      
                                                                                                                                         
  assert.Nil(t, quick.Check(propertyTest, nil))                                                                                          
}
```

Reply:
caught a bug in my code, make sure the `bytes.Compare` compares against `truncatedInput` instead of the possibly longer `input`

Reply:
https://github.com/tendermint/go-wire/issues/63